pango/src/glyph_item_iter.rs

```diff
@@ -1,63 +1,186 @@
 // Take a look at the license at the top of the repository in the LICENSE file.
 
 use crate::GlyphItem;
-use crate::GlyphItemIter;
-use glib::translate::*;
-use std::mem;
+use glib::{translate::*, GStr};
+use std::{marker::PhantomData, mem};
 
-impl GlyphItemIter {
+#[derive(Clone, Debug)]
+#[repr(transparent)]
+pub struct GlyphItemIter<'item, 'text> {
+    inner: ffi::PangoGlyphItemIter,
+    item: PhantomData<&'item GlyphItem>,
+    text: PhantomData<&'text str>,
+}
+
+impl<'item, 'text> glib::StaticType for GlyphItemIter<'item, 'text> {
+    #[inline]
+    fn static_type() -> glib::Type {
+        unsafe { from_glib(ffi::pango_glyph_item_iter_get_type()) }
+    }
+}
+
+impl<'item, 'text> GlyphItemIter<'item, 'text> {
+    #[doc(alias = "pango_glyph_item_iter_next_cluster")]
+    pub fn next_cluster(&mut self) -> bool {
+        unsafe {
+            from_glib(ffi::pango_glyph_item_iter_next_cluster(
+                self.to_glib_none_mut().0,
+            ))
+        }
+    }
+
+    #[doc(alias = "pango_glyph_item_iter_prev_cluster")]
+    pub fn prev_cluster(&mut self) -> bool {
+        unsafe {
+            from_glib(ffi::pango_glyph_item_iter_prev_cluster(
+                self.to_glib_none_mut().0,
+            ))
+        }
+    }
     #[doc(alias = "pango_glyph_item_iter_init_end")]
-    pub fn new_end(glyph_item: &GlyphItem, text: &str) -> Result<Self, glib::BoolError> {
+    pub fn new_end(
+        glyph_item: &'item GlyphItem,
+        text: impl AsRef<GStr> + 'text,
```

Comment:
Do we want to change this here to `AsRef<GStr>` already, or defer that to next release with the other related changes?

Reply:
We can, but either way if you want to remove `AsRef<GStr>` then the `GlyphItemIter` Rust struct will have to be able to store a string. When we do `IntoGStr` then it will have to be generic on that

Reply:
Changed to store a `GString` for now